### PR TITLE
Modify parallel hashagg spill doc

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

**Review by joey-yez — APPROVED** [review]

File: configure-memory-usage.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ TiDB 支持对执行算子的数据落盘功能。当 SQL 的内存使用超过
 
 - 落盘行为由参数 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query)、[`tidb_enable_tmp_storage_on_oom`](/system-variables.md#tidb_enable_tmp_storage_on_oom)、[`tmp-storage-path`](/tidb-configuration-file.md#tmp-storage-path)、[`tmp-storage-quota`](/tidb-configuration-file.md#tmp-storage-quota) 共同控制。
 - 当落盘被触发时，TiDB 会在日志中打印一条包含关键字 `memory exceeds quota, spill to disk now` 或 `memory exceeds quota, set aggregate mode to spill-mode` 的日志。
-- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并发 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并发 HashAgg 的落盘是从 v8.0.0 版本开始引入的。
+- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并发 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并发 HashAgg 的落盘是从 v8.0.0 版本开始引入，在 v8.2.0 正式 GA。用户可以通过系统变量`tidb_enable_parallel_hashagg_spill`来关闭和打开并发 HashAgg 的落盘功能(这是临时系统变量，未来会被删除)。
```
Comment (Oreoxmt) [inline configure-memory-usage.md:156]:
```suggestion
- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并发 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并发 HashAgg 的落盘在 v8.0.0 版本以实验特性引入，在 v8.2.0 版本成为正式功能 (GA)。你可以通过系统变量 [`tidb_enable_parallel_hashagg_spill`](/system-variables.md#tidb_enable_concurrent_hashagg_spill-从-v800-版本开始引入) 控制是否启用支持落盘的并发 HashAgg 算法。该变量将在未来版本中废弃。
```

Remove the warning in L1783-L1885:

https://github.com/pingcap/docs-cn/blob/ce3458ca27e646a6ad6098b8f14c9bfa0755c442/system-variables.md?plain=1#L1781-L1792

File: configure-memory-usage.md [hunk 2]
```diff
@@ -153,7 +153,7 @@ TiDB 支持对执行算子的数据落盘功能。当 SQL 的内存使用超过
...
 - 当包含 Sort、MergeJoin、HashJoin 或 HashAgg 的 SQL 语句引起内存 OOM 时，TiDB 默认会触发落盘。
```
Comment (Oreoxmt) [inline configure-memory-usage.md:158]: - In v8.0.0, this variable is introduced with an incorrect name `tidb_enable_concurrent_hashagg_spill` (ref: https://github.com/pingcap/docs-cn/pull/15789). Please create a new PR to rename it to `tidb_enable_parallel_hashagg_spill`.
- Please confirm whether this variable is deprecated in v8.2.0.

Comment (xzhangxian1008) [reply configure-memory-usage.md:158]: > * In v8.0.0, this variable is introduced with an incorrect name `tidb_enable_concurrent_hashagg_spill` (ref: [Spill of parallel aggregation in TiDB #15789](https://github.com/pingcap/docs-cn/pull/15789)). Please create a new PR to rename it to `tidb_enable_parallel_hashagg_spill`.
> * Please confirm whether this variable is deprecated in v8.2.0.

This variable is not deprecated in v8.2.0, but user will get a warning when he sets this variable.

File: system-variables.md [hunk 3]
```diff
@@ -2175,16 +2175,12 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
->
-> 该变量控制的功能为实验特性，不建议在生产环境上使用。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
-
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`ON`
-- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，并行 HashAgg 将支持落盘。该变量将在功能正式发布后废弃。
+- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，并行 HashAgg 将支持落盘。从 v8.2.0 开始，当你将该变量设置为`OFF`的时候会得到一个 warning。该变量在未来版本中会被完全废弃。
```
Comment (Oreoxmt) [inline system-variables.md:2183]:
```suggestion
- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，并行 HashAgg 将支持落盘。从 v8.2.0 开始，将该变量设置为 `OFF` 时会产生警告。该变量将在未来版本中完全废弃。
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -2175,16 +2175,12 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
->
-> 该变量控制的功能为实验特性，不建议在生产环境上使用。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
-
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`ON`
-- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，并行 HashAgg 将支持落盘。该变量将在功能正式发布后废弃。
+- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，并行 HashAgg 将支持落盘。从 v8.2.0 开始，将该变量设置为 `OFF` 时会产生警告。该变量将在未来版本中完全废弃。
```
Comment (qiancai) [inline system-variables.md:2183]: 不确定是否可以说下 `ON` 的好处，比如下面这样，避免直接说 "设置为 `OFF` 时会产生警告"有点突兀

```suggestion
- 这个变量用来控制 TiDB 是否支持并行 HashAgg 进行落盘。当该变量设置为 `ON` 时，在任意并发条件下，HashAgg 算子都可以根据内存使用情况自动触发数据落盘，从而兼顾性能和数据处理量。因此，不推荐将此变量修改为 `OFF`。从 v8.2.0 开始，将该变量设置为 `OFF` 时会产生警告。该变量将在未来版本中废弃。
```

Comment (xzhangxian1008) [reply system-variables.md:2183]: > 不确定是否可以说下 `ON` 的好处，比如下面这样，避免直接说 "设置为 `OFF` 时会产生警告"有点突兀

+1

**Review by qiancai — APPROVED** [review]

File: configure-memory-usage.md [hunk 5]
```diff
@@ -153,7 +153,7 @@ TiDB 支持对执行算子的数据落盘功能。当 SQL 的内存使用超过
 
 - 落盘行为由参数 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query)、[`tidb_enable_tmp_storage_on_oom`](/system-variables.md#tidb_enable_tmp_storage_on_oom)、[`tmp-storage-path`](/tidb-configuration-file.md#tmp-storage-path)、[`tmp-storage-quota`](/tidb-configuration-file.md#tmp-storage-quota) 共同控制。
 - 当落盘被触发时，TiDB 会在日志中打印一条包含关键字 `memory exceeds quota, spill to disk now` 或 `memory exceeds quota, set aggregate mode to spill-mode` 的日志。
-- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并行 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并行 HashAgg 的落盘是从 v8.0.0 版本开始引入的。
+- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并行 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并行 HashAgg 的落盘在 v8.0.0 版本以实验特性引入，在 v8.2.0 版本成为正式功能 (GA)。你可以通过系统变量 [`tidb_enable_parallel_hashagg_spill`](/system-variables.md#tidb_enable_concurrent_hashagg_spill-从-v800-版本开始引入) 控制是否启用支持落盘的并行 HashAgg 算法。该变量将在未来版本中废弃。
```
Comment (Oreoxmt) [inline configure-memory-usage.md:156]:
```suggestion
- Sort、MergeJoin、HashJoin 落盘是从 v4.0.0 版本开始引入的，非并行 HashAgg 的落盘是从 v5.2.0 版本开始引入的，并行 HashAgg 的落盘在 v8.0.0 版本以实验特性引入，在 v8.2.0 版本成为正式功能 (GA)。你可以通过系统变量 [`tidb_enable_parallel_hashagg_spill`](/system-variables.md#tidb_enable_parallel_hashagg_spill-从-v800-版本开始引入) 控制是否启用支持落盘的并行 HashAgg 算法。该变量将在未来版本中废弃。
```

**Pull request merged**